src/views/AddItem.jsx

```diff
@@ -1,36 +1,60 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import Form from '../components/form';
 import { addItem } from '../api';
+import { isCompositeComponent } from 'react-dom/test-utils';
 
 export function AddItem({ listToken }) {
 	const defaultFormValues = {
 		itemName: '',
 		daysUntilNextPurchase: 7,
-		isSubmited: false,
+		isSubmitted: false,
+		isLoading: false,
+		isSuccess: false,
 	};
 
 	const [form, setForm] = useState(defaultFormValues);
 
-	const handleSubmit = (event) => {
+	const handleSubmit = async (event) => {
 		event.preventDefault();
 		if (form.itemName.length === 0) {
 			return;
 		}
-		setForm({ ...defaultFormValues, isSubmited: true });
-		addItem(listToken, form);
+		setForm({ ...defaultFormValues, isSubmitted: true, isLoading: true });
+		const { success, error } = await addItem(listToken, form);
+		success
+			? setForm({
+					...defaultFormValues,
+					isSubmitted: true,
+					isSuccess: true,
+					isLoading: false,
+			  })
+			: setForm({
+					...defaultFormValues,
+					isSubmitted: true,
+					isSuccess: false,
+					isLoading: false,
+					error,
+			  });
```

Comment:
According to [this portion of the React docs](https://beta.reactjs.org/learn/state-as-a-snapshot) the state does not update until the component re-renders. This appears to be what is making setting the state back to default values above sending the form's state to the database not cause an issue, because state still has the original values until after the re-render. 

In office hours yesterday we discussed moving the call to set the form back to default to be after the addItem call for clearing up how it reads. However, based on the new isLoading state needing to be set above the addItem call I don't think that plan will work. Everything appears to be working so I think leaving this how it is written is the best option, but I just wanted to make a note so we can look back at why it was kept this way if we need to in the future.